stan/inc2prev.stan

```diff
@@ -50,95 +53,108 @@ data {
+    // inc_init is the mean incidence
+    infections[i] = inv_logit(init_inc[i] + gp[i]);
 
-  // calculate detectable cases
-  dcases = convolve(infections, prob_detect);
-  // calculate observed detectable cases
-  odcases = observed_in_window(dcases, prev_stime, prev_etime, ut, obs);
-  //combined standard error
-  combined_sigma = sqrt(square(sigma) + prev_sd2);
+    // calculate detectable cases
+    dcases[i] = to_row_vector(convolve(to_vector(infections[i]), prob_detect));
+    // calculate observed detectable cases
+    odcases[i] = to_row_vector(observed_in_window(to_vector(dcases[i]), prev_stime, prev_etime, ut, obs));
+    //combined standard error
+    combined_sigma[i] = sqrt(rep_row_vector(square(sigma), obs) + prev_sd2[i]);
+  }
 
   //calculate infections with potential to have antibodies
-  if (ab_obs) {
-    infs_with_potential_abs = convolve(infections, inf_ab_delay);
+  for (i in 1:n_ab) {
+    infs_with_potential_abs[i] = to_row_vector(convolve(to_vector(infections[ab_index[i]]), inf_ab_delay));
     // calculate detectable antibodies
-    dab = detectable_antibodies(infs_with_potential_abs, vacc_with_ab, beta[1],
-                                gamma, delta[1], k[1], l[1], init_dab[1], t);
+    dab[i] = to_row_vector(detectable_antibodies(to_vector(infs_with_potential_abs[i]),
+                                                 to_vector(vacc_with_ab[ab_index[i]]), 
+                                                 beta[1], gamma, delta[1], k[1], l[1],
+                                                 init_dab[ab_index[i]], t));
     // calculate observed detectable antibodies
-    odab = observed_in_window(dab, ab_stime, ab_etime, ut, ab_obs);
+    odab[i] = to_row_vector(observed_in_window(to_vector(dab[i]), ab_stime, ab_etime, ut, ab_obs));
     //combined standard error
-    combined_ab_sigma = sqrt(square(ab_sigma[1]) + ab_sd2);
+    combined_ab_sigma[i] = sqrt(rep_row_vector(square(ab_sigma[1]), ab_obs) + ab_sd2[i]);
   }
 }
 
 model {
   // gaussian process priors
-  rho ~ inv_gamma(lengthscale_alpha, lengthscale_beta);
-  alpha ~ std_normal() T[0,];
-  eta ~ std_normal();
+  for (i in 1:n) {
+    rho[i] ~ inv_gamma(lengthscale_alpha, lengthscale_beta);
+    alpha[i] ~ std_normal() T[0,];
+    for (j in 1:M) {
+      eta[i, j] ~ std_normal();
+    }
+    init_inc[i] ~ normal(init_inc_mean[i], 2);
+  }
 
   // Initial infections
-  init_inc ~ normal(init_inc_mean, 2);
-  if (diff_order) {
-    init_growth ~ normal(0, 0.25);
+  for (i in 1:n) {
+    for (j in 1:diff_order) {
+      init_growth[i, j] ~ normal(0, 0.25);
+    }
   }
+
   // prevalence observation model
   for (i in 1:pbt) {
     prob_detect[i] ~ normal(prob_detect_mean[i], prob_detect_sd[i]) T[0, 1];
   }
 
   // Priors for antibody model
-  if (ab_obs) {
-    init_dab ~ normal(init_ab_mean[1], init_ab_sd[1]);
+  if (n_ab > 0) {
+    init_dab ~ normal(init_ab_mean, init_ab_sd);
     logit(beta) ~ normal(pbeta[1], pbeta[2]);
```

Comment:
Note to self that need to update how this is handled as I think as written requires a jacobian adjustment.